tests/unit/test_charm.py

```diff
@@ -1,18 +1,1264 @@
+            networks={"juju-info": Network.default()},
+            leader=True,
+        )
+
+        with patch(
+            "requests.get",
+            return_value=Mock(
+                **{"json.return_value": {"initialized": True}, "status_code": 200},
+            ),
+        ):
+            context.run(Event("config-changed"), state)
+
+    def test_given_only_config_storage_container_can_connect_network_available_gocert_initialized_when_configure_then_no_error_raised(
+        self, context
+    ):
+        state = State(
+            storage=[Storage(name="config")],
+            containers=[Container(name="gocert", can_connect=True)],
+            networks={"juju-info": Network.default()},
+            leader=True,
+        )
+
+        with patch(
+            "requests.get",
+            return_value=Mock(
+                **{"json.return_value": {"initialized": True}, "status_code": 200},
+            ),
+        ):
+            context.run(Event("config-changed"), state)
+
+    def test_given_only_database_storage_container_can_connect_network_available_gocert_initialized_when_configure_then_no_error_raised(
+        self, context
+    ):
+        state = State(
+            storage=[Storage(name="database")],
+            containers=[Container(name="gocert", can_connect=True)],
+            networks={"juju-info": Network.default()},
+            leader=True,
+        )
+
+        with patch(
+            "requests.get",
+            return_value=Mock(
+                **{"json.return_value": {"initialized": True}, "status_code": 200},
+            ),
+        ):
+            context.run(Event("config-changed"), state)
+
+    def test_given_storages_available_container_can_connect_network_available_gocert_initialized_when_configure_then_status_is_active(
+        self, context
+    ):
+        state = State(
+            storage=[Storage(name="config"), Storage(name="database")],
+            containers=[Container(name="gocert", can_connect=True)],
+            networks={"juju-info": Network.default()},
+            leader=True,
+        )
+
+        with patch(
+            "requests.get",
+            return_value=Mock(
+                **{"json.return_value": {"initialized": True}, "status_code": 200},
+            ),
+        ):
+            context.run(Event("config-changed"), state)
+
+    # Test Unit Status
+    def test_given_only_config_storage_container_cant_connect_network_not_available_gocert_not_running_when_collect_status_then_status_is_waiting(
+        self, context
+    ):
+        state = State(
+            storage=[Storage(name="config")],
+            containers=[Container(name="gocert", can_connect=False)],
+            networks={"juju-info": Network([], [], [])},
+            leader=True,
+        )
+
+        with patch("requests.get", return_value=Mock(side_effect=requests.ConnectionError)):
+            out = context.run(Event("config-changed"), state)
```

Comment:
```suggestion
            out = context.run(Event("collect_unit_status"), state)
```

This comment applies to every test for collect unit status

Reply:
already changed